core/src/validator.rs

```diff
@@ -168,7 +170,44 @@ impl Validator {
             config.storage_slots_per_turn,
             bank.slots_per_segment(),
         );
-
+        let bf = bank_forks.clone();
+        let kp = keypair.clone();
+        let mut counter = 0;
+        let counter_ref = &mut counter;
+        if counter == 0 {
```

Comment:
I think we want this counter inside the `set_handler()`.  It'll always be `0` here since Validator::new() is only ever called once.  

The point of `counter` is to signal an exit on the 2nd Ctrl+c right?   I don't see any code that checks the value of `counter` (or `counter_ref`) and actually causes an exit.

Reply:
I thought the point of the counter is to warn a validator about stake deactivation upon first press of Ctrl+c and gracefully terminate upon the second press of CTRL+C. That's why I put our counter outside of set_handler() to enforce execution of set_handler only upon first press of Ctrl+C . 

" It'll always be 0 here since Validator::new() is only ever called once."  That's true. i think maybe we have to move set_handler to another place

Reply:
I think the current location of `set_handler()` is fine.  It's just that the closure just needs to check whether it's the first or second time it's been executed as it gets executed on every Ctrl+C right?